Subject: [PATCH] smbios spoofing
---
Index: OvmfPkg/SmbiosPlatformDxe/SmbiosPlatformDxe.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/OvmfPkg/SmbiosPlatformDxe/SmbiosPlatformDxe.c b/OvmfPkg/SmbiosPlatformDxe/SmbiosPlatformDxe.c
--- a/OvmfPkg/SmbiosPlatformDxe/SmbiosPlatformDxe.c	(revision f9408b7cc160030c3bb9627c8086a792cc2d046c)
+++ b/OvmfPkg/SmbiosPlatformDxe/SmbiosPlatformDxe.c	(date 1749744080699)
@@ -38,9 +38,15 @@
   },
   {      // BIOSCharacteristicsExtensionBytes[2]
     0,   // BiosReserved
-    0x1C // SystemReserved = VirtualMachineSupported |
-    //                  UefiSpecificationSupported |
-    //                  TargetContentDistributionEnabled
+    //
+    // SystemReserved:
+    // - UefiSpecificationSupported (Bit 4)
+    // - TargetContentDistributionEnabled (Bit 3)
+    //
+    // The VirtualMachineSupported flag (Bit 2) has been deliberately
+    // cleared to bypass VM detection mechanisms. Original value was 0x1C.
+    //
+    0x18
   },
   0,     // UINT8                     SystemBiosMajorRelease
   0,     // UINT8                     SystemBiosMinorRelease
@@ -191,4 +197,4 @@
   }
 
   return EFI_SUCCESS;
-}
+}
\ No newline at end of file
